torch/distributed/_composable_state.py

```diff
@@ -0,0 +1,24 @@
+from typing import cast, Dict, Optional
+
+import torch.nn as nn
+
+
+class _State:
+    pass
+
+
+_module_state_mapping: Dict[nn.Module, _State] = {}
+
+
+def _insert_module_state(module: nn.Module, state: _State) -> None:
+    global _module_state_mapping
+    assert module not in _module_state_mapping, f"Inserting {module} more than once."
+    _module_state_mapping[module] = state
+
+
+def _get_module_state(module: nn.Module) -> Optional[_State]:
+    global _module_state_mapping
+    if isinstance(module, _State):
+        return cast(_State, module)
+    else:
+        return _module_state_mapping.get(module, None)
```

Comment:
nit: Do you think this might be clearer? (up to you)
```suggestion
def _get_module_state(module_or_state: Union[nn.Module, _State]) -> Optional[_State]:
    global _module_state_mapping
    if isinstance(module_or_state, _State):
        return cast(_State, module_or_state)
    else:
        return _module_state_mapping.get(module_or_state, None)
```

Reply:
This is semantically not correct. The API is used to map from `module` to `state` where `state` can be either a module (module-wrapped case) or `_FSDPSTate`. So the input argument must be a module.

Reply:
Ohh ok. I understand what is going on now. `FullyShardedDataParallel` inherits from `_FSDPState`, which inherits from `_State`, so wrapper FSDP is the `if isinstance(module, _State)` code path.